pandas/io/formats/style_render.py

```diff
@@ -124,6 +128,7 @@ def __init__(
         self.hide_columns_: list = [False] * self.columns.nlevels
         self.hidden_rows: Sequence[int] = []  # sequence for specific hidden rows/cols
         self.hidden_columns: Sequence[int] = []
+        self.descriptors: list[str | Callable | tuple[str, Callable]] = []
```

Comment:
Is this a common API?  I think most places where alternative input are allowed either use a Sequence of some sort or `dict[str, str | Callable]`.  Would be be cleaner if the allowed inputs were `Sequence[str | Callable] | dict[str | Callable]`.  

Also, can't the `Callable be better described?  Isn't is `Callable[[Series], float]`?  It seems that the `Callable` must take a `Series` and return some sort of scalar value.  Thie question is whether the scalar value can be non-numeric.  If it could be any numeric value, than you could you use something like `Callable[[Series], int | float]`.  So more generally, what are the requirements for the `Callable`?

Reply:
> 
> Also, can't the `Callable be better described? Isn't is `Callable[[Series], float]`? It seems that the `Callable`must take a`Series`and return some sort of scalar value. Thie question is whether the scalar value can be non-numeric. If it could be any numeric value, than you could you use something like`Callable[[Series], int | float]`. So more generally, what are the requirements for the `Callable`?

The Callable could techincally return anything so long as the returned object has a `__str__` representation to populate in the HTML element. Common values might be int, float, string, but yes it does accept only series.

Reply:
I think the sig should then be `Callable[[Series],Any]`

Reply:
- [x] changed typing. But did not add the dict. `List[Tuple[str, str]]` is present for some of the `Styler` functions